lib/chef/provider/git.rb

```diff
@@ -323,6 +319,12 @@ def git(*args)
         ["git", *args].compact.join(" ")
       end
 
+      def run_git_command(args, run_opts = {})
+        git_command = git(args)
```

Comment:
I'd actually suggest pulling the args mangling directly into here and then renaming `git_run_command` to just `git` -- the two methods don't really add a lot and the verbose naming could be dropped and `git("ls-remote", ...)` is pretty readable and understandable.